pkg/pillar/types/zedmanagertypes.go

```diff
@@ -172,6 +178,8 @@ type AppInstanceStatus struct {
 	// All error strings across all steps and all StorageStatus
 	// ErrorAndTimeWithSource provides SetError, SetErrrorWithSource, etc
 	ErrorAndTimeWithSource
+	// The flag says the application has already been started being processed after the delay timeout is passed
+	DelayHandled bool
```

Comment:
Why do you need this separate boolean? 
Can't you tell that the start time has already passed or that the state is not START_DELAY?

Reply:
The problem is when the checkDelayedStartApps function triggers doUpdate, it takes time to change the state of the domain from START_DELAYED to something else. So, on the next run of the checkDelayedStartApps function (that can happen quite soon, as we have the ticker set to 1 sec), it can trigger doUpdate for the domain that is already being handled. To eliminate it, I added an extra flag.

Reply:
Nope, it's set to booting only at the very end of the doActivate() function.